backend/routes/auth.routes.js

```diff
@@ -0,0 +1,62 @@
+const bcrypt = require('bcrypt');
+const express = require('express');
+const jwt = require('jsonwebtoken')
+const passport = require('passport');
+
+const db = require('../model');
+
+const router = express.Router();
+
+// Register
+router.post('/register', async (req, res) => {
+
+  const { username, email, password } = req.body;
+
+  if(!email || !username || !password){
+    return res.status(400).send({ success: false, message: 'Username, email, and password are required.' });
+  }
+
+  // check if the email already exists
+  const existingEmail = await db.user.findOne({ where: { email } });
+  if (existingEmail) {
+    return res.status(409).send({success: false, message: 'Email already exists' });
```

Comment:
One question I also have is whether or not we want to indicate whether an email already exists on the platform. I have read conflicting things about this practice.

Reply:
I'm open to changing this to allow one email to sign up for multiple accounts. Usernames would still be unique?

Reply:
I'm not sure about the email signing up for multiple accounts - more like changing the message so that it doesn't specifically state that that email already exists?
[OWASP Auth Guidelines](https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/Authentication_Cheat_Sheet.md#authentication-and-error-messages)